### DISCUSSION TITLE
Where are the WM_ constants?

### DISCUSSION THREAD

**Discussion by Slion** [microsoft/CsWin32#497] [Q&A] [answered]
Is there a way to generate the window message constants?
I was looking for WM_SETICON just now but any or all of them would be great.

**Comment by AArnott** [comment, marked as answer]
Yes, just specify `WM_SETICON` in the NativeMethods.txt file. Then you can discover where it was generated by pressing Ctrl+T while your caret is still on that API name.
By default it is generated here in the latest CsWin32 version:

```cs
namespace Windows.Win32
{
	using global::System;
	using global::System.Diagnostics;
	using global::System.Runtime.CompilerServices;
	using global::System.Runtime.InteropServices;
	using winmdroot = global::Windows.Win32;

	internal static partial class PInvoke
	{
		internal const uint WM_SETICON = 128U;
	}
}
```
